src/utils/addons/prepare.js

```diff
@@ -0,0 +1,110 @@
+const chalk = require('chalk')
+
+const ADDON_VALIDATION = {
+  EXISTS: 'EXISTS',
+  NOT_EXISTS: 'NOT_EXISTS',
+}
+
+const validateExists = ({ addon, addonName, siteData, log, exit }) => {
+  if (!addon || !addon.id) {
+    log(`Add-on ${addonName} doesn't exist for ${siteData.name}`)
+    log(`> Run \`netlify addons:create ${addonName}\` to create an instance for this site`)
+    exit(1)
+  }
+}
+
+const validateNotExists = ({ addon, addonName, siteData, log, exit }) => {
+  if (addon && addon.id) {
+    log(`The "${addonName} add-on" already exists for ${siteData.name}`)
+    log()
+    const cmd = chalk.cyan(`\`netlify addons:config ${addonName}\``)
+    log(`- To update this add-on run: ${cmd}`)
+    const deleteCmd = chalk.cyan(`\`netlify addons:delete ${addonName}\``)
+    log(`- To remove this add-on run: ${deleteCmd}`)
+    log()
+    exit(1)
+  }
+}
+
+const getCurrentAddon = ({ addons, addonName }) => {
+  const addon = addons.find(addon => addon.service_path === `/.netlify/${addonName}`)
+  return addon
+}
+
+const validateCurrentAddon = ({ addon, validation, addonName, siteData, log, warn, exit }) => {
+  switch (validation) {
+    case ADDON_VALIDATION.EXISTS: {
+      validateExists({ addon, addonName, siteData, log, exit })
+      break
+    }
+    case ADDON_VALIDATION.NOT_EXISTS: {
+      validateNotExists({ addon, addonName, siteData, log, exit })
+      break
+    }
+    default: {
+      warn(`Unknown addons validation: ${validation}`)
+      break
+    }
+  }
+}
+
+const getAddonManifest = async ({ api, addonName, error }) => {
+  let manifest
+  try {
+    manifest = await api.showServiceManifest({ addonName })
+  } catch (e) {
+    if (typeof e.message === 'string' && e.message.includes('Not Found')) {
+      error(`No add-on "${addonName}" found. Please double check your add-on name and try again`)
+    } else {
+      error(e.message)
+    }
+  }
+  return manifest
+}
+
+const getSiteData = async ({ api, siteId, error }) => {
+  let siteData
+  try {
+    siteData = await api.getSite({ siteId })
+  } catch (e) {
+    error(`Failed getting list of site data: ${e.message}`)
+  }
+  return siteData
+}
+
+const getAddons = async ({ api, siteId, error }) => {
+  let addons
+  try {
+    addons = await api.listServiceInstancesForSite({ siteId })
+  } catch (e) {
+    error(`Failed getting list of addons: ${e.message}`)
```

Comment:
Should we handle `422` like in https://github.com/netlify/js-client/blob/00b3e939ef133a15db0e9ba8e4e873d748fd442b/src/addons.js#L41?

Reply:
Oh, didn't realise we're returning the status code of the error:
https://github.com/netlify/micro-api-client/blob/c08716a0d83c4e826bca99ab2a081015c98ebf44/src/index.js#L4
Going to add that back

Reply:
Ha, we don't to handle that error code, `js-client` throws an error in that case:
https://github.com/netlify/js-client/blob/00b3e939ef133a15db0e9ba8e4e873d748fd442b/src/methods/response.js#L10

Reply:
Great. 
It's actually better because it handles non-JSON responses (which sometimes happen due to bugs), and it includes the full response for JSON responses.